### ISSUE TITLE
Refactor the read line function

### ISSUE THREAD

**Issue by ThakeeNathees** [ThakeeNathees/pocketlang#79]
## Re-implement [readLine()](https://github.com/ThakeeNathees/pocketlang/blob/5f6f66517fbc3779e3db7960ab05d8fa76246a3f/cli/repl.c#L19-L30) function

The pocketlang's `input()` function and `cli`'s input reading both calls the `read_line()` function defined in `cli/repl.c` which is something I wrote for temporary. that can only read at a maximum of 1024 characters and not safe. Which needs to be refactored with reading each character one by one and write it to a buffer till we reach a new line or an EOF.

## How to implement.

- Create **another** function with the name `readLine` (and don't refactor the `read_line` function because it's used by the VM's IO callbacks, and refactoring it is for the future). That'll take a byte buffer and read the chars into that buffer. (here is an un tested code)
```c
static void readLine(ByteBuffer* buff) {
  do {
    char c = fgetc(stdin);
    if (c == EOF || c == '\n') break;

    byteBufferWrite(buff, (uint8_t)c);
  } while (true);

  byteBufferWrite(buff, '\0');
}
```
- Don't forget to write the comments documenting the function.
    - What it does at above the function's declaration.
    - How it does inside the function, at required places (But here we don't need any).

- Create a buffer to store the line that'll be read from stdin.
```diff
+ // A buffer to store a line read from stdin.
+  ByteBuffer line;
+  byteBufferInit(&line);

   // A buffer to store lines read from stdin.
   ByteBuffer lines;
   byteBufferInit(&lines);
```

- Remove the `read_line()` call and change it to `readLine(&line)`

- Change all the `line` with `(const char*)line.data`

- Change all the `free((void*)line);` with `byteBufferClear(&line);`

## Finally

Unit tests for REPL haven't implemented yet, so you have to run some tests by yourself to make sure everything works fine, before making the PR. (ex:)

```
>>> def hello()
...     print('hello world')
... end
>>>
>>> hello()
hello world
>>>
```

**Comment by ThakeeNathees** [comment]
Note that `line.count` will return the element count of the buffer including the null byte `\0`, use `line.count - 1` for getting the length of the line.

CC @xSavitar

**Comment by xSavitar** [comment]
Will take a look at this shortly. Thanks!

**Comment by ThakeeNathees** [comment]
Fixed by #83
